rpcclient/wsclient/wsclient.go

```diff
@@ -255,26 +255,35 @@ func (c *wsClient) TakeOfferAndSubscribe(multiaddr, offerID string,
 		return nil, err
 	}
 
-	// read ID from connection
-	message, err := c.read()
-	if err != nil {
-		return nil, fmt.Errorf("failed to read websockets message: %s", err)
-	}
+	readMessageFn := func() (string, error) {
```

Comment:
`TakeOfferAndSubscribe` is 172 lines lines. Maybe we could pull this code out into it's own function and then the code auditor doesn't have to carefully read it to understand which variables are shared with the outer scope?

Reply:
updated!